src/standalone/Vegetation/Canopy.jl

```diff
@@ -461,6 +537,14 @@ function CanopyModel{FT}(;
         @assert typeof(boundary_conditions.atmos) <: PrescribedAtmosphere{FT}
     end
 
+    if typeof(photosynthesis) <: PModel{FT}
+        @assert typeof(conductance) <: PModelConductance{FT} "When using PModel for photosynthesis, you must also use PModelConductance for stomatal conductance"
+    end
+
+    if typeof(conductance) <: PModelConductance{FT}
+        @assert typeof(photosynthesis) <: PModel{FT} "When using PModelConductance for stomatal conductance, you must also use PModel for photosynthesis"
+    end
```

Comment:
A more concise way to write this check:
```suggestion
(photosynthesis isa PModel{FT} && conductance isa PModelConductance{FT}) ||
    @error("The PModel option must be used either for both photosynthesis and conductance, or for neither.")
```

Reply:
oops yeah it should be:
```suggestion
(photosynthesis isa PModel{FT} && conductance isa PModelConductance{FT}) ||
    (!(photosynthesis isa PModel{FT}) && !(conductance isa PModelConductance{FT}))
    @error("The PModel option must be used either for both photosynthesis and conductance, or for neither.")
```